crates/project/src/lsp_store.rs

```diff
@@ -8316,6 +8353,36 @@ impl PartialEq for LanguageServerPromptRequest {
     }
 }
 
+/// A show document request by LSP server
+#[derive(Clone, Debug)]
+pub struct LanguageServerShowDocumentRequest {
+    pub uri: Url,
+    pub external: bool,
```

Comment:
There are a few other optional parameters that I've chosen to ignore - https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#showDocumentParams

Reply:
Why don't we add the others so that we're on-spec